File: java/src/test/java/TemplateMethodTest.java
import org.junit.jupiter.api.Test;
import template_method.PostAccountController;
import template_method.PostAccountReqBody;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class TemplateMethodTest {

    @Test
    void testTemplateMethod() {
        final var controller = new PostAccountController();
        final var response = controller.createAccount(new PostAccountReqBody("12345", "123"));
        assertEquals(201, response.status());
    }
}

File: src/test/kotlin/TemplateMethodTest.kt
import io.kotest.core.spec.style.FunSpec
import io.kotest.matchers.shouldBe
import template_method.PostAccountController
import template_method.PostAccountReqBody

class TemplateMethodTest : FunSpec({
    test("Test Template Method") {
        val controller = PostAccountController()
        val response = controller.createAccount(PostAccountReqBody("12345", "123"))
        response.status shouldBe 201
    }
})